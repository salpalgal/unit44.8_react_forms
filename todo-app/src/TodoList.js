import React , {useState} from "react"
import Todo from "./Todo";
import NewTodoForm from "./NewTdoForm";
import {v4 as uuid} from "uuid"

const TodoList = ()=>{
    const initial = [{task: ""}];

    const [tasks, setTasks] = useState(initial);

    const addTask = (newTask)=>{
        setTasks(tasks=>[...tasks,{...newTask,id:uuid()}])
    }
    const removeTask = id =>{
        setTasks(tasks=> tasks.filter(task=>task.id !== id))
    }
    const allTasks = tasks.map(({task,id})=>
    <Todo task={task} id={id} removeTask={removeTask}/>)

    return (
        <div>
            <h3>Todo List</h3>
           <NewTodoForm addTask={addTask}/>
           {allTasks}
        </div>
    )
}

export default TodoList ; 