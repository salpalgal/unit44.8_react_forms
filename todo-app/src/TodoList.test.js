import React from "react";
import { render ,fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";

function addTask(todolist, task="walk dog"){
    const taskInput = todolist.getByLabelText("Task")
    fireEvent.change(taskInput, { target: { value: task } });
    const button = todolist.getByText("Add")
    fireEvent.click(button)
}

it("renders without crashing", function() {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add task", function(){
    const todolist = render(<TodoList />)
    addTask(todolist)
    const removeBtn = todolist.getByText("X")
    expect(removeBtn).toBeInTheDocument()

})

it("can remove task", function(){
    const todolist = render(<TodoList/>)
    addTask(todolist)
    const removeBtn = todolist.getByText("X")
    fireEvent.click(removeBtn)
    expect(removeBtn).not.toBeInTheDocument()
})
