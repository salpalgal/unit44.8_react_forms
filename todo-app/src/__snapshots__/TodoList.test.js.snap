// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="Enter task"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
